src/builder/QuarkBuilder.sol

```diff
@@ -1160,52 +1209,11 @@ contract QuarkBuilder {
                 revert InvalidActionChain();
             }
             paymentTokenCost += nonBridgeAction.paymentMaxCost;
-
-            if (Strings.stringEqIgnoreCase(nonBridgeAction.actionType, Actions.ACTION_TYPE_BORROW)) {
-                continue;
-            } else if (Strings.stringEqIgnoreCase(nonBridgeAction.actionType, Actions.ACTION_TYPE_REPAY)) {
-                Actions.RepayActionContext memory cometRepayActionContext =
-                    abi.decode(nonBridgeAction.actionContext, (Actions.RepayActionContext));
-                if (Strings.stringEqIgnoreCase(cometRepayActionContext.assetSymbol, paymentTokenSymbol)) {
-                    paymentTokenCost += cometRepayActionContext.amount;
-                }
-            } else if (Strings.stringEqIgnoreCase(nonBridgeAction.actionType, Actions.ACTION_TYPE_SUPPLY)) {
-                Actions.SupplyActionContext memory cometSupplyActionContext =
-                    abi.decode(nonBridgeAction.actionContext, (Actions.SupplyActionContext));
-                if (Strings.stringEqIgnoreCase(cometSupplyActionContext.assetSymbol, paymentTokenSymbol)) {
-                    paymentTokenCost += cometSupplyActionContext.amount;
-                }
-            } else if (Strings.stringEqIgnoreCase(nonBridgeAction.actionType, Actions.ACTION_TYPE_SWAP)) {
-                Actions.SwapActionContext memory swapActionContext =
-                    abi.decode(nonBridgeAction.actionContext, (Actions.SwapActionContext));
-                if (Strings.stringEqIgnoreCase(swapActionContext.inputAssetSymbol, paymentTokenSymbol)) {
-                    paymentTokenCost += swapActionContext.inputAmount;
-                }
-            } else if (Strings.stringEqIgnoreCase(nonBridgeAction.actionType, Actions.ACTION_TYPE_TRANSFER)) {
-                Actions.TransferActionContext memory transferActionContext =
-                    abi.decode(nonBridgeAction.actionContext, (Actions.TransferActionContext));
-                if (Strings.stringEqIgnoreCase(transferActionContext.assetSymbol, paymentTokenSymbol)) {
-                    paymentTokenCost += transferActionContext.amount;
-                }
-            } else if (
-                Strings.stringEqIgnoreCase(nonBridgeAction.actionType, Actions.ACTION_TYPE_UNWRAP)
-                    || Strings.stringEqIgnoreCase(nonBridgeAction.actionType, Actions.ACTION_TYPE_WRAP)
-            ) {
-                Actions.WrapOrUnwrapActionContext memory wrapOrUnwrapActionContext =
-                    abi.decode(nonBridgeAction.actionContext, (Actions.WrapOrUnwrapActionContext));
-                if (Strings.stringEqIgnoreCase(wrapOrUnwrapActionContext.fromAssetSymbol, paymentTokenSymbol)) {
-                    paymentTokenCost += wrapOrUnwrapActionContext.amount;
-                }
```

Comment:
I think once this logics are removed, so I assume all this logics will be handled separately in above steps. 
So probably need to add this to each wrapOrUnwrap step as well. 
(When wrap/unwrap away from payment token (pay with ETH, wrapping ETH away to WETH), the cost need to adjust accordingly)